journal.py

```diff
@@ -139,40 +144,37 @@ def logout(request):
 def add_entry(request):
     if request.authenticated_userid:
         try:
-            write_entry(request)
-        except psycopg2.Error:
-            # This will catch any errors generated by the database
+            Entry.from_request(request)
+        except SQLAlchemyError:
             return HTTPInternalServerError
-        cursor = request.db.cursor()
-        cursor.execute(NEWEST_ENTRY)
-        keys = ('id', 'title', 'text', 'created')
-        entry = dict(zip(keys, cursor.fetchone()))
-        entry['text'] = markdown.markdown(entry['text'], extensions=['codehilite', 'fenced_code'])
-        entry['created'] = entry['created'].strftime('%b. %d, %Y')
-        return entry
+        new = Entry.recent()
+        json_add = {
+            'id': new.id,
+            'title': new.title,
+            'text': markdown.markdown(new.text,
+                                      extensions=['codehilite', 'fenced_code']),
+            'created': new.created.strftime('%b. %d, %Y')
+        }
+        return json_add
     else:
         return HTTPForbidden()
 
 
 @view_config(route_name='home', renderer='templates/list.jinja2')
 def read_entries(request):
-    cursor = request.db.cursor()
-    cursor.execute(READ_ENTRIES)
-    keys = ('id', 'title', 'text', 'created')
-    entries = [dict(zip(keys, row)) for row in cursor.fetchall()]
+    entries = Entry.all()
     for e in entries:
-        e['text'] = markdown.markdown(e['text'], extensions=['codehilite', 'fenced_code'])
+        e.text = markdown.markdown(e.text,
+                                   extensions=['codehilite', 'fenced_code'])
```

Comment:
Be careful here.  The value bound to `e` is a database-connected entry object.  Mutating its `text` attribute like this means that you've changed the text that will be stored in the database to be *rendered* html instead of raw markdown.  Because of automatic transaction management, that change is picked up by the DBSession and it will be committed.